src/view/Mafs.tsx

```diff
@@ -113,13 +107,96 @@ export function Mafs({
   const viewBoxX = round((xMin / (xMax - xMin)) * width, 2)
   const viewBoxY = round((yMax / (yMin - yMax)) * height, 2)
 
+  useGesture(
+    {
+      onDrag: ({ movement: [mx, my], memo = camera, type, event }) => {
+        if (type.includes("key")) event.preventDefault()
+
+        setCamera(
+          vec
+            .matrixBuilder(memo)
+            .translate((-mx / width) * xSpan, (my / height) * ySpan)
+            .get()
+        )
+
+        return memo
+      },
+      onPinch: ({ movement: [scale], origin, event, memo = 1 }) => {
+        if (!event.currentTarget || !inverseViewTransform) return 1
+
+        const min = 0.1
+        const max = 5
+
+        // Bail out early if the camera is already at the zoom boundaries.
+        if ((camera[0] === max && scale < 1) || (camera[0] === min && scale > 1)) return scale
```

Comment:
There still some kinks to work out here—zooming out has some very slight jank when you hit the boundary. It might be cool to have some kind of rubber-banding effect.